internal/httputil/errors.go

```diff
@@ -10,14 +10,16 @@ import (
 	"github.com/pomerium/pomerium/internal/templates"
 )
 
-// HTTPError stores the status code and a message for a given HTTP error.
-type HTTPError struct {
-	Code    int
-	Message string
+// Error reports an http error, its http status code, a custom message, and
+// whether it is troubleshootable.
+type Error struct {
+	Message          string
+	Code             int
+	Troubleshootable bool
```

Comment:
Nit: Maybe canTroubleshoot or canDebug, unless that term is threaded through the codebase already.